### ISSUE TITLE
ocropus-gpageseg: Too many column separators detected

### ISSUE THREAD

**Issue by lehzwo** [tmbdev/ocropy#171]
## Current Behavior
When passing _n_ as a value for the maximum number of white column separators via `--maxcolseps`, _n+1_ white column separators are detected instead. Therefore it is not possible to process an image without taking at least one white column separator into account.
The detection of black column separators via `-b` on the other hand detects the correct number of black separators if the passed value of `--maxseps ` does not equal 0. In this case all black separators are detected.

## Possible Solution

At _line 212_ in _compute_colseps_conv_ in _ocropus-gpageseg_
`seps = morph.select_regions(seps,sl.dim0,min=args.csminheight*scale,nbest=args.maxcolseps+1)`,
`args.maxcolseps+1` is passed instead of `args.maxcolseps`. Therefore an additional whitespace separator is selected.

When changing that line to 
`seps = morph.select_regions(seps,sl.dim0,min=args.csminheight*scale,nbest=args.maxcolseps)`,the whitespace separator detection behaves like the black separator detection mentioned above (if 0 is passed, all separators are detected). This can be prevented by adding the check `if nbest > 0:` to the block 
```
for i in best[-nbest:]:
   if scores[i]<=min: continue
   keep[i+1] = 1
```
at line 198 in _select_regions_ in _morph.py_

**Comment by zuphilip** [comment]
@lehzwo Thank you for this report and I have also seen your PR. I will need more time to look at it, think about and try out some things... 

Do you have an example image which shows this behavior and how your PR solves the issue? I am not sure but I have something in mind, that the a white column separator is also needed for the "left border" and therefore even in a single-column text. This might be related to https://github.com/tmbdev/ocropy/issues/115.

The change might be critical for anyone who relies on the "old behavior". Are there alternatives which are extending the old behavior with the new functionality?

Let me also ping @jze and @1a1a11a which worked on tables in ocropy before. Any reactions from you?

**Comment by lehzwo** [comment]
I processed the image [http://dibiki.ub.uni-kiel.de/viewer/fullscreen/PPN828360804/32/](url) via _ocropus-nlbin_ without additional parameters to get the binary image. Without the changes of the PR, the following command
`ocropus-gpageseg -n -d --maxcolseps 0`
detects one white column separator in the middle of the image. This separator is taken into account when creating the line seeds.

![pre_pr_colwsseps](https://cloud.githubusercontent.com/assets/23117070/21992688/b3c9df1c-dc17-11e6-94de-c11773eac587.png)
![pre_pr_lineseeds](https://cloud.githubusercontent.com/assets/23117070/21992681/af19fbb4-dc17-11e6-9fe6-02fd311d6289.png)

with the changes of the PR, the white separator is ignored.

![pr_colwsseps](https://cloud.githubusercontent.com/assets/23117070/21992766/0536ecc8-dc18-11e6-9d2d-455ea60a46f3.png)
![pr_lineseeds](https://cloud.githubusercontent.com/assets/23117070/21992767/06404038-dc18-11e6-8acd-9f418cf725ab.png)

I'm interested in the possibility of ignoring column separators because @jze had the idea of applying a mask to an image, if the column separator detection did not work as expected. The mask can be used to specify the column separators of the page that have not been detected. In some cases the user might be interested in specifying all separators on his own, because detected column separators like the one above might decline the result like in the example above, where the last letter of the word "Polizeibezirk" is ignored and the second last one is damaged. In these cases it would be nice to simply set --maxcolseps to 0 to only apply the separators of the mask and prevent the automatic detection.  

The only way I can currently think of to prevent the automatic detection of whitespace separators without changing the "old behavior" would be an additional parameter to prevent the detection of whitespace separators at all. But this might lead to confusion because of the inconsistent behavior of the parameters concerning the column separators where one parameter prevents the detection of whitespace separators and the _-b_ separator on the other hand allows the detection of black separators.

**Comment by zuphilip** [comment]
Solved by #172. Thank you very much for the detailed description here and the PR.

I begin to document some parameters in the wiki, see https://github.com/tmbdev/ocropy/wiki/Page-Segmentation#whitespace-column-separators
